cmd/sourced/compose/workdir/factory.go

```diff
@@ -167,6 +212,41 @@ func (f *envFile) String() string {
 	`, f.Workdir, volumeType, volumeSource, gitbaseSiva,
 		strings.Join(f.GithubOrganizations, ","), f.GithubToken, noForks,
 		gitbaseLimitCPU, gitcollectorLimitCPU, gitbaseLimitMem)
+
+	return []byte(result), nil
+}
+
+func (f *envFile) UnmarshalEnv(b []byte) error {
+	r := bytes.NewReader(b)
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		if strings.TrimSpace(scanner.Text()) == "" {
+			continue
+		}
+
+		line := strings.SplitN(scanner.Text(), "=", 2)
+		value := line[1]
```

Comment:
I think we should check the length of `line`.
A line like `# A comment` will panic.

It should not fail with the current contents, but just in case we may change the contents later, or a user may edit this manually...

Reply:
fixed by skipping lines without `=`, thanks